base/set.jl

```diff
@@ -631,7 +643,7 @@ julia> replace!(isodd, A, 0, count=2)
 ```
 """
 replace!(pred::Callable, A, new; count::Integer=typemax(Int)) =
-    replace!(x -> ifelse(pred(x), new, x), A, count=count)
+    replace!(x -> ifelse(pred(x), new, x), A, count=check_count(count))
```

Comment:
This `replace!` method already checks `count`, so not really needed to use `check_count` here (or did you mean to save one specialization when `count` is not an `Int` ?)

Reply:
Yes, I added it to avoid the specialization and to throw the error as early as possible. But maybe that's not worth it...